sdk/core/azure-core-amqp/src/main/java/com/azure/core/amqp/implementation/ReactorSession.java

```diff
@@ -162,6 +180,117 @@ public Duration getOperationTimeout() {
         return openTimeout;
     }
 
+    /**
+     *
+     * @return {@link Mono} of {@link TransactionCoordinator}
+     */
+    private Mono<TransactionCoordinator> createTransactionCoordinator() {
+        if (isDisposed()) {
+            return Mono.error(logger.logExceptionAsError(new IllegalStateException(String.format(
+                "Cannot create coordinator send link '%s' from a closed session.", TRANSACTION_LINK_NAME))));
+        }
+
+        if (transactionCoordinator != null) {
+            logger.verbose("Coordinator[{}]: Returning existing transaction coordinator.", TRANSACTION_LINK_NAME);
+            return Mono.just(transactionCoordinator);
+        }
+
+        synchronized (coordinatorLock) {
```

Comment:
1. Why is a lock here? The operation itself is asynchronous. The call returns immediately after setting up the Mono. You can have multiple createCoordinatorSendLink.
1. Creating an AtomicReference _plus_ a lock for this doesn't seem like the right approach here.